src/backend/distributed/worker/worker_data_fetch_protocol.c

```diff
@@ -611,10 +609,13 @@ worker_append_table_to_shard(PG_FUNCTION_ARGS)
 	uint64 shardId = ExtractShardIdFromTableName(shardTableName, false);
 	LockShardResource(shardId, AccessExclusiveLock);
 
-	/* copy remote table's data to this node */
+	/*
+	 * Copy into intermediate results directory, which is automatically cleaned on
+	 * error.
+	 */
 	StringInfo localFilePath = makeStringInfo();
-	appendStringInfo(localFilePath, "base/%s/%s" UINT64_FORMAT,
-					 PG_JOB_CACHE_DIR, TABLE_FILE_PREFIX, shardId);
+	appendStringInfo(localFilePath, "%s/worker_append_table_to_shard_" UINT64_FORMAT,
+					 CreateIntermediateResultsDirectory(), shardId);
```

Comment:
`CreateIntermediateResultsDirectory()` is coupled  with distributed transactions -- when we are in a dist transaction.

One thing I'm trying to understand is could have have a case where we call `CreateIntermediateResultsDirectory` when we are not in a dist. transaction, and call `RemoveIntermediateResultsDirectory` when we are in a dist. transaction.

I'm thinking about this because we call  `worker_append_table_to_shard ` liberally, sometimes in a tx sometime outside tx.

As far as I can see, there is no such issue, do you see any regarding `CreateIntermediateResultsDirectory()`?

Reply:
You could probably construct such a scenario manually, but ultimately this is an internal function called by master_append_table_to_shard, which uses assign_distributed_transaction_id.